pkg/kubelet/kubelet.go

```diff
@@ -3310,6 +3457,14 @@ func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.P
 	return *s
 }
 
+// convertStatusToAPIStatus creates an api PodStatus for the given pod from
+// the given internal pod status.  It is purely transformative and does not
+// alter the kubelet state at all:
+// 1.  get the old container statuses
+// 2.  create new container statuses in the default waiting state, and project fields from the old statuses onto them
+// 3.  sort the statuses so the most recent come first
+// 4.  set the container statuses based on the given internal pod status.
+// 5.  set the status correctly of containers waiting to start
```

Comment:
Do we really need so much implementation details upfront? I am suspicious that this would help people understand the code better, but I could easily be wrong. 

Besides, the code may also diverge from the comment easily...